### PR TITLE
feat!: Allow configuring port numbers and validate config properties

### PR DESCRIPTION
**Pull request by sbernauer**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,8 +4,13 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Added
+
+- Added a configuration to specify the port numbers for `http`, `https` and `metrics` ([#XX]).
```
Comment (NickLarsenNZ) [inline CHANGELOG.md:9]:
```suggestion
- Added a configuration to specify the port numbers for `http`, `https` and `metrics` ([#43]).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,8 +4,13 @@ All notable changes to this project will be documented in this file.
...
+
 ### Changed
 
+- BREAKING: Ensure no unknown config properties have been set. This is to make the user aware that what he tried to configure is not a valid configuration. You may need to adopt your configuration and remove any unknown properties ([#XX]).
```
Comment (NickLarsenNZ) [inline CHANGELOG.md:13]:
```suggestion
- BREAKING: Ensure no unknown config properties have been set. This is to make the user aware that what he tried to configure is not a valid configuration. You may need to adapt your configuration and remove any unknown properties ([#43]).
```

Comment (NickLarsenNZ) [reply CHANGELOG.md:13]: You'll need to add `[#43]: https://github.com/stackabletech/trino-lb/pull/43`, as the line is beyond where I can make a suggestion in Github Issues.

**Review by NickLarsenNZ — APPROVED** [review]
Very nice, just need to add the PR link in the changelog

File: CHANGELOG.md (hunk 2 above)
Comment (NickLarsenNZ) [reply CHANGELOG.md:13]: The suggestion wasn't applied

Comment (sbernauer) [reply CHANGELOG.md]: Sorry, I only saw the PR number. typo is fixed

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Pull request merged**